src/components/challenges/complete.tsx

```diff
@@ -30,20 +28,46 @@ type CompleteProps = {
   };
 };
 
-const Complete: FC<CompleteProps> = ({ content }) => (
-  <div className={classNames.boxContent}>
-    <p className={classNames.heading}>{content.title}</p>
-    <Details text="Level" value={content.level} />
-    <Details text="Challenge Started" value={content.start_date} />
-    <Details text="Challenge Ends" value={content.end_date} />
-    <div className={classNames.participants}>
-      <Details text="Participants" value={content.participants.length} />
-      <Participants participants={content.participants} />
-    </div>
-    <p className={classNames.viewStats}>
-      <a href="/">View Stats</a>
-    </p>
-  </div>
-);
+const Complete: FC<CompleteProps> = ({ content }) => {
+  const taskData = {
+    Level: content.level,
+    Challenge_Started: content.start_date,
+    Challenge_Ends: content.end_date,
+    Active_Participants: content.participants.length,
+  };
+
+  const task: any[] = [];
+  function getTask() {
+    // eslint-disable-next-line array-callback-return
+    Object.entries(taskData).map(([key, value]) => {
+      task.push({ key, value });
+    });
+    return task;
+  }
+
+  const arrayOfParticipants: any[] = content.participants;
+
+  const participants: any[] = [];
+  function getParticipants(users: any[]) {
+    // eslint-disable-next-line array-callback-return
+    users.map((user) => {
+      participants.push({
+        userName: user.rds_member_id,
+        imgUrl: `${process.env.NEXT_PUBLIC_GITHUB_IMAGE_URL}${user.rds_member_id}/img.png`,
+        key: user.rds_member_id,
+      });
+    });
+    return participants;
+  }
```

Comment:
Same code repeated on file `src/components/challenges/active.tsx`
How about refactoring this into one component and using that component on both these pages?

Reply:
Refactored this code into separate files (details and participants) to avoid repetition.

https://github.com/Real-Dev-Squad/website-status/commit/f84f9afa7df2679012784eb4f1b1558a2496fdb4